README.md

```diff
@@ -17,240 +20,154 @@ of the [Privacy Community Group](https://privacycg.github.io/).
+
+```javascript
+// Records that the user is logging-in to a FedCM-compliant Identity Provider.
+navigator.setLoggedIn({
+  idp: true
+});
 ```
 
-For the promise to resolve, the user needs to already have login
-status set for the federated login provider, i.e. the user needs to
-be logged in to the provider first.
+##### Storage Access API
+
+The [Storage Access API](https://github.com/privacycg/storage-access/issues/8#issue-560633211) could benefit from a login status signal by allowing developers to call the API with the option to only show user-facing prompts if the user is logged into their site, and reject otherwise. The Storage Access API must avoid directly exposing the login status to attackers that are measuring the time to rejection, by delaying the rejection by random response times that reflect average real user response times (which are usually in the range of a few seconds).
 
-Then the destination website has to call the API on its side:
+```javascript
+// Records that the user is logging-in, which allows the Storage Access API to conditionally dismiss its
+// UX.
+navigator.setLoggedIn();
 
+// In another cross-site top-level document, auto-reject rSA (with some delay to avoid timing attacks) unless the user is logged in
+document.requestStorageAccess({rejectUnlessLoggedIn: true}).then(...);
 ```
-navigator.recordFederatedLogin(
-    loginProvider: secure origin,
-    username,
-    credentialTokenType,
-    optionalParams { }
-) –> Promise<void>
+
+##### Browser Status UI
+
+Much like favicons, websites may be benefited from having a login status indicator in browser UI (e.g. the URL bar). The indicator could extend the login status API to gather the explicit signals that it needs from the website to display the indicator (e.g. a name and an avatar).
+
+```javascript
+// Records that the user is logging-in to a FedCM-compliant Identity Provider.
+navigator.setLoggedIn({
+  profile: {
+    name: "John Doe",
+    picture: "https://website.com/john-doe/profile.png",
+  }
+});
 ```
 
-The promise would only resolve if the `loginProvider` had recently
-called `initiateRecordFederatedLogin()` for this destination website.
+#### The Assumption of Abuse
 
-## Challenges and Open Questions
+Every user of the login status (e.g. Web Standards or browser features integrating with it) **MUST** incorporate into their threat model that it is:
+
+1. Self-declared: any website can and will lie to gain any advantage
+2. Client-side: the state represent the website's client-side knowledge of the user's login status, which is just an approximation of the server-side's state (which is the ultimate source of truth) 
+
+One potential for abuse is if websites don’t call the logout API when they should. This could allow them to maintain the privileges tied to login status even after the user logged out.
```

Comment:
What about the potential for abuse if sites call the login API when the user is not actually logging in?

Reply:
The specific use being proposed as the first extension of the Login Status API is for FedCM (and possibly the Storage Access API) to use the signal.

In both of these cases, the consumers (FedCM/SAA) have to construct the use of the signal in such a way that assume the signal is self-declared. So, for FedCM specifically, we show an "Error/mismatch UI" when the information we get from (a) the Login Status API and (b) the [accounts endpoint](https://fedidcg.github.io/FedCM/#idp-api-accounts-endpoint) returns. The user experience of the Error UI is constructed in such a way that a valid / honest Identity Provider wouldn't want to abuse itself (by lying to the browser), and an dishonest IdP would just show Error UIs to users. 

The same occurs for dishonest/honest Identity Providers lying about the user logging-out.

We have constructed the extension of the Login Status API such that a website self-declares its status as an Identity Provider:

```javascript
// Records that the user is logging-in to a FedCM-compliant Identity Provider.
navigator.setLoggedIn({
  idp: true
});
```

The `idp: true` parameter indicates that the website is declaring itself as a FedCM-compatible IdP (which is not game-able). Any other Web Platform API that wanted to consume this signal would have to either (a) have to work within the FedCM assumptions of abuse or (b) extend the Login Status API.

For example, @johannhof is making the case that the SSA API can take this signal as is, so could reuse this signal with:

```javascript
// Records that the user is logging-in, which allows the Storage Access API to conditionally dismiss its
// UX.
navigator.setLoggedIn();

// In another cross-site top-level document, auto-reject rSA (with some delay to avoid timing attacks) unless the user is logged in
document.requestStorageAccess({rejectUnlessLoggedIn: true}).then(...);
```

The signal doesn't have to always be self-declared, it can be mediated by the user too. So, for example, we could extend the Login Status API to have a browser prompt (or perhaps just a status indicator), that would make the change visible enough that a browser would be able to trust it as acknowledged by the user.

```javascript
navigator.setLoggedIn({
  profile: {
    name: "John Doe",
    picture: "https://website.com/john-doe/profile.png",
  }
});
```

Reply:
We _could_ do those things, but shouldn't trustworthiness considerations in the Explainer cover the whole span of envisioned use cases, not just the FedCM one?

Reply:
It does cover use cases that are designed in a way where site-declared login status can safely provide user benefits. Maybe it makes sense to call this a "base" or "untrusted" tier and have another tier for "trusted" applications of the API?